packages/coinjoin/src/utils/http.ts

```diff
@@ -79,11 +79,18 @@ export const coordinatorRequest = async <R = void>(
         try {
             response = await httpPost(`${baseUrl}${url}`, body, { ...options, signal });
         } catch (e) {
-            if ('code' in e && e.code === 'ECONNRESET') {
-                // catch errors from nodejs http module like "socket hang up" or "socket disconnected before secure TLS connection was established" etc.
-                // When sending request through a keep-alive enabled agent the underlying socket might be reused.
-                // But if server closes connection at unfortunate time client may run into a 'ECONNRESET' error.
-                // each case explanation in ./node_modules/@types/node/*/http.d.ts
+            // NOTE: this code probably belongs to @trezor/request-manager package since errors are tightly related to TOR
+            // catch errors from:
+            // - nodejs http module like "socket hang up" or "socket disconnected before secure TLS connection was established" (see ./node_modules/@types/node/*/http.d.ts)
+            // - socks module like "Socks5 proxy rejected connection" or "Proxy connection timed out" (see ./node_modules/socks/build/common/constants)
+            const socksErrors = ['Socks5', 'Proxy'];
+            const shouldSwitchIdentity =
+                ('code' in e && e.code === 'ECONNRESET') ||
+                ('type' in e &&
+                    e.type === 'system' &&
+                    socksErrors.find(se => e.message.includes(se)));
```

Comment:
ignorable nitpick: probably "righter" to use `socksError.some` instead of `socksError.find`

Reply:
i've removed socksErrors completely relying only on e.type === 'system' now

Reply:
ah :( relying only on `type: system` will not work.
`ECONNREFUSED` is also using it while this is a "valid" error  (server `https://localhost/status` does not exist)
```
FetchError {
      message: 'request to https://localhost/status failed, reason: connect ECONNREFUSED ::1:443',
      type: 'system',
      errno: 'ECONNREFUSED',
      code: 'ECONNREFUSED'
}
```